#! /usr/bin/env python

PACKAGE='mrpt_rbpf_slam'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("folder_simplemap",    str_t,    0, "folder where to save *.simplemap",  "/tmp")
gen.add("update_while_stopped", bool_t, 0, "keep updating filter while the robot is stopped", True)
gen.add("update_sensor_pose", bool_t, 0, "keep updating the sensor pose using the tf tree", False)

gen.add("visualization_on", bool_t, 0, "On true it opens the mrpt viewer", True)
gen.add("visualization_update_delay", int_t, 0, "window update delay", 1, 0, 10)
gen.add("visualization_width", int_t, 0, "window width", 600, 320, 1920)
gen.add("visualization_height", int_t, 0, "height", 500, 280, 1080)
gen.add("visualization_follow_robot", bool_t, 0, "Camera follows the robot", True)

exit(gen.generate(PACKAGE, "mrpt_rbpf_slam", "General"))
